src/library/scala/collection/mutable/LinkedHashMap.scala

```diff
@@ -129,6 +132,59 @@ class LinkedHashMap[K, V]
     }
   }
 
+  override def filterInPlace(p: (K, V) => Boolean): this.type = {
+    if (size > 0) {
+      var bucket = 0 // the index of the current bucket in the hash table
+      val tableTable = table.table
+      val buckets = tableTable.length
+      var newFirstEntry: Entry = null // will become the new firstEntry
+      var earlier: Entry = null // the entry "before" the current entry, may cross bucket boundaries
+      var newTableSize = table.tableSize
+
+      while (bucket < buckets) {
+        (tableTable(bucket) : @switch) match {
+          case null => ()
+          case head =>
+            var newHead: Entry = null // the new beginning of this bucket
+          var prev: Entry = null // the previous node within this bucket, does not cross bucket boundaries
+          var next: Entry = head.asInstanceOf[Entry] // the next node within this bucket, does not cross bucket boundaries
+
+            while (next ne null) {
```

Comment:
I think this also should be split into prefix and postfix loops as I described for `HashMap`.

Reply:
:+1: